src/Access/Authentication.cpp

```diff
@@ -8,6 +8,15 @@
 #include <Poco/SHA1Engine.h>
 #include <Common/typeid_cast.h>
 
+extern "C" {
+#include <hmac.h>
```

Comment:
What do you think of using Poco implementation https://docs.pocoproject.org/current/Poco.HMACEngine.html ?